service.iptv.manager/resources/lib/kodilogging.py

```diff
@@ -8,37 +8,51 @@
 import xbmc
 import xbmcaddon
 
+from resources.lib import kodiutils
+
+ADDON = xbmcaddon.Addon()
+
 
 class KodiLogHandler(logging.StreamHandler):
-    """A log handler for Kodi"""
+    """ A log handler for Kodi """
 
     def __init__(self):
         logging.StreamHandler.__init__(self)
-        addon_id = xbmcaddon.Addon().getAddonInfo("id")
-        formatter = logging.Formatter("[{}] [%(name)s] %(message)s".format(addon_id))
+        formatter = logging.Formatter("[{}] [%(name)s] %(message)s".format(ADDON.getAddonInfo("id")))
         self.setFormatter(formatter)
+        # xbmc.LOGNOTICE is deprecated in Kodi 19 Matrix
+        if kodiutils.kodi_version_major() > 18:
+            self.info_level = xbmc.LOGINFO
+        else:
+            self.info_level = xbmc.LOGNOTICE
```

Comment:
The min. debug level ought to be xbmc.LOGDEBUG to avoid spamming logs.

Reply:
That is just defining the info_level as it is different between Kodi 18 and 19. Not actual logging

Reply:
@Gujal00 if you evaluate the full code you'll see this sets the min. log-level when debug is enabled within the plugin. Essentially a Kodi debug switch bypass. I have already discussed this with the developer and since its set only when a user allows it... It's okay for submission.